main.py

```diff
@@ -208,41 +199,124 @@ def get_flows_to_be_installed(self):
                     # Change table_id before being added
                     flow["flow"].update({"table_id": expected_table_id})
                     install_flows[switch].append(flow["flow"])
+
         log.info(f"of_multi_table pushing flows, pipeline: {pipeline}")
+        self.manage_miss_flows(pipeline, flows_by_swich)
         self.send_flows(delete_flows, "delete")
         self.send_flows(install_flows, "install")
+
         if pipeline.get("status") is None:
             self.pipeline_controller.disabled_pipeline(pipeline_id)
             msg = f"Pipeline {pipeline_id} disabled"
             log.debug(f"of_multi_table result {msg}")
         else:
-            self.install_miss_flows(pipeline)
             self.pipeline_controller.enabled_pipeline(pipeline_id)
             msg = f"Pipeline {pipeline_id} enabled"
             log.debug(f"of_multi_table result {msg}")
 
-    def install_miss_flows(self, pipeline: dict):
+    @staticmethod
+    def get_miss_flows_installed(
+        flows_by_swich: Dict[str, Dict]
+    ) -> tuple[dict[int, dict], set[str]]:
+        """Get miss flows reformated as {"table_id": {flow}} from
+        a single switch"""
+        miss_flows = {}
+        stored_table_ids = set()
+        # Keys needed to compare miss flow entries, except table_id
+        compare = {"priority", "instructions", "match"}
+        for _, flows in flows_by_swich.items():
+            for flow in flows:
+                owner = flow["flow"].get("owner")
+                if owner == "of_multi_table":
+                    flow = flow["flow"]
+                    miss_flows[flow["table_id"]] = {
+                        key: flow[key] for key in compare if flow.get(key) is not None
+                    }
+                    stored_table_ids.add(flow["table_id"])
+            # Miss flows are the same in all switches
+            break
+        return miss_flows, stored_table_ids
+
+    def manage_miss_flows(self, pipeline: Dict, flows_by_swich: Dict[str, Dict]):
+        """Determine whether to install and/or delete miss_flows"""
+        if not self.controller.switches:
```

Comment:
In case where `coloring`, `of_lldp` or `telemetry_int` are not installed. At the start of kytos, there are probably no flows installed but a pipeline with miss flows is enabled. The miss flows only need switches to be installed.

Reply:
@Alopalao, this seems like it can lead to a potential issue. Let's verify by parts:

* `coloring`, `of_lldp`, `mef_eline` and future `telemetry_int` are expected to be dependencies and loaded first than `of_multi_table`. We'll structure the docker image (for prod usage, and in development they should follow the same ordering) and recommend it accordingly. As you mentioned, if any flows miss might (or might not) be there, so maybe we don't even need a early return here at all since your diff algo will do what's expected? 

    * Otherwise it might discard an important execution of FlowMods of flow misses (even if the switches might not be connected since we're leveraging `force=true`). Also if a switch has ever existed and it if flows were installed, `topology` will always guarantee that this switch will be loaded first since `topology` has a higher priority in our docker file since most NApps depends on it, and currently we don't support switch hard deletions, so a concern of potentially sending to a non existing switch isn't an issue as far as I can follow.

* If any of these NApps aren't installed, is a valid concern for this iteration, but shouldn't this even be treated way earlier? For instance, before even trying [to enable the pipeline on load_pipeline?](https://github.com/kytos-ng/of_multi_table/blob/master/main.py#L76-L81) since at that point `handle_enable_table` would be stuck since a reply might never come? As it currently is, seems fair game to me though, since eventually these NApps that are missing can also get loaded, maybe a warning message on `load_pipeline` method should suffice if any of the subscribed NApps aren't loaded? and then we can keep `load_pipeline` pretty much as it is but with with an exception warning message if that ever happens?

Reply:
Let me know if you're seeing something else differently or if you have other suggestions

Reply:
Oh ye, I can just remove it. But for completeness this condition was just for a very specific case:
- Only `mef_eline` is installed/enabled.
- There are no EVCs deployed, therefore no flows.
- A pipeline was enabled and it has miss flow entries in it.
In this case the request GET from `http://localhost:8181/api/kytos/flow_manager/v2/stored_flows?state=installed` is going to return an empty dictionary and the previous condition was going to do an early return withou installing the miss flows.

Reply:
E2E tests results without the condition
```
============================= test session starts ==============================
platform linux -- Python 3.9.2, pytest-7.2.0, pluggy-1.0.0
rootdir: /tests
plugins: timeout-2.1.0, rerunfailures-10.2, anyio-3.6.2
collected 5 items

tests/test_e2e_60_of_multi_table.py .....                                [100%]
```